### DISCUSSION TITLE
lammps examples for nvt and npt simulations

### DISCUSSION THREAD

**Discussion by deleep225** [mir-group/nequip#156] [General]
I am trying to do the nvt simulations with trained models. I found wild fluctuations in temperatures.

I have also taken the google-colab tutorial and tried to run the simulation in nvt at 298 K. I found similar wild fluctuation in temperature.

Step       PotEng                KinEng            TotEng                Temp                  Press             Volume
       0   -169816.55           12.43594            -169804.11          298                   0.56847511      1000000
     100   -169797.52        14.312021           -169783.2          342.95619         0.65423509      1000000
     200   -169804.61       42.009572            -169762.6          1006.6672          1.9203532      1000000
     300   -169778.45       4.4903736e-108   -16977 8.45       1.0760195e-106 2.052652e-109      1000000
     400   -169816.11       330.76314            -169485.35         7926.0129          15.119936      1000000
     500    -169816.5        2.6527385e-86    -169816.5           6.3567057e-85    1.2126272e-87      1000000
     600   -169814.08       2.3112458e-07   -169814.08         5.5383935e-06     1.0565231e-08      1000000
     700   -169816.41      0.018748718        -169816.39        0.44927188          0.00085704658      1000000
     800   -169806.48      25.813136          -169780.67           618.55515            1.1799772      1000000
     900   -169788.61     51.641929         -169736.97            1237.4855            2.3606702      1000000
    1000   -169781.58    8.1885765         -169773.39          196.22127           0.37431847      1000000
    1100   -169787.17   1.3127544e-09   -169787.17        3.1457279e-08        6.0008994e-11      1000000
    1200   -169740.02    79.204829        -169660.81          1897.9699                3.620633      1000000

from the above output you can see high fluctuation in temperature  and kinetic energy.

here in my lammps input file

units   real
atom_style atomic
newton off
thermo 1
read_data em1

pair_style      nequip
pair_coeff      * * toluene-deployed.pth C H
mass            1 15.9994
mass            2 1.00794

neighbor 1.0 bin
neigh_modify delay 5 every 1

velocity        all create 298.0 23456789

fix             1 all nvt temp 298.0 298.0 0.5 #iso 1.0 1.0 0.5
#fix 2 all langevin  298.0 298.0 298.0 5646546
timestep        1
thermo_style    custom step pe ke etotal temp press vol
thermo          100
#dump            1 all custom 100 water1ns.dump id type x y z
dump 1 all atom 100 2.lammpstrj
run             20000
write_data      nvt1

It will be great if some lammps examples are provided in nvt, npt and nve ensembles with nequip potentials.

Thank you.

**Comment by simonbatzner** [comment]
HI @deleep225, your Tdamp parameter is 0.5 and you are in real units, that means it's a 0.5fs Tdamp for a 1fs time step. That won't work. [The recommendation is](https://docs.lammps.org/fix_nh.html): `$(100*dt)`

**Comment by simonbatzner** [comment]
Note also that for systems with C-H bonds, 0.5fs is typically used as a safe choice for a time step due to the high frequency of the C-H bond.
